src/prices/PythCollector.ts

```diff
@@ -0,0 +1,187 @@
+import { HermesClient, HexString, PriceUpdate } from '@pythnetwork/hermes-client';
+import { Cell, Dictionary } from '@ton/core';
+import { checkNotInDebtAtAll } from '../api/math';
+import { packPythUpdatesData } from '../api/prices';
+import { FeedMapItem, OracleConfig, parseFeedsMapDict, PoolAssetConfig, PoolAssetsConfig } from '../types/Master';
+import { FetchPricesConfig, Oracle } from './Oracle.interface';
+import { Prices } from './Prices';
+import { PythFeedUpdateType, PythPriceSourcesConfig } from './Types';
+
+export type PythCollectorConfig = {
+    poolAssetsConfig: PoolAssetsConfig;
+    pythOracle: OracleConfig;
+    pythConfig: PythPriceSourcesConfig;
+};
+
+export class PythCollector implements Oracle {
+    #oracleInfo: OracleConfig;
+    #parsedFeedsMap: Map<bigint, FeedMapItem>;
+    #pythConfig: PythPriceSourcesConfig;
+    #poolAssetsConfig: PoolAssetsConfig;
+
+    constructor(config: PythCollectorConfig) {
+        this.#oracleInfo = config.pythOracle;
+        this.#pythConfig = config.pythConfig;
+        this.#poolAssetsConfig = config.poolAssetsConfig;
+        this.#parsedFeedsMap = parseFeedsMapDict(this.#oracleInfo.feedsMap);
+    }
+
+    async getPricesForLiquidate(
+        realPrincipals: Dictionary<bigint, bigint>,
+        fetchConfig: FetchPricesConfig,
+    ): Promise<Prices> {
+        const assets = this.#filterEmptyPrincipalsAndAssets(realPrincipals);
+        if (assets.includes(undefined)) {
+            throw new Error('User from another pool');
+        }
+        return await this.getPrices(
+            assets.map((x) => x!),
+            fetchConfig,
+        );
+    }
+
+    async getPricesForSupplyWithdraw(
+        realPrincipals: Dictionary<bigint, bigint>,
+        supplyAsset: PoolAssetConfig | undefined,
+        withdrawAsset: PoolAssetConfig | undefined,
+        collateralToDebt: boolean,
+        fetchConfig: FetchPricesConfig,
+    ): Promise<Prices> {
+        let assets = this.#filterEmptyPrincipalsAndAssets(realPrincipals);
+        if (
+            checkNotInDebtAtAll(realPrincipals) &&
+            withdrawAsset &&
```

Comment:
```suggestion
            !withdrawAsset &&
```
не уверен, что именно так, но логика что он не задан